internal/app/orchestrator/orchestrator.go

```diff
@@ -166,6 +167,23 @@ func (o *orchestrator) CreateWatch(req transport.Request) (transport.Watch, func
 	}
 	metrics.OrchestratorWatchSubscope(o.scope, aggregatedKey).Counter(metrics.OrchestratorWatchCreated).Inc(1)
 
+	// Log + stat to investigate NACK behavior
+	resourceString := ""
+	if req.GetResourceNames() != nil {
+		resourceString = strings.Join(req.GetResourceNames()[:], ",")
+	}
+	if isNackRequest(req) {
+		o.logger.With(
+			"request_version", req.GetVersionInfo(),
+			"resource_names", resourceString,
```

Comment:
`resourceString` seems to be used inside the if condition. May be good to move it closer to usage